private-templates-service/adaptivecards-templating-service/src/util/mongoutils/mongoutils.ts

```diff
@@ -16,4 +17,37 @@ export module MongoUtils {
     }
     return result;
   }
+
+  export function restoreJSONTypeOfTemplate(templateModels: ITemplateModel[]): ITemplateModel[] {
+    return templateModels.map(template => {
+      template.instances = template.instances.map(instance => {
+        instance.json = MongoUtils.stringToJSON(Object.assign(instance.json) as string);
+        if (instance.data) {
+          instance.data = MongoUtils.stringArrayToJSONArray(Object.assign(instance.data) as string[]);
+        }
+        return instance;
+      });
+      return template;
+    });
```

Comment:
error handling?

Reply:
shouldn't be any, in case of empty array => empty array gonna be returned. Json by default is always there